src/Quadrature.jl

```diff
@@ -43,6 +43,58 @@ function scale_x(ub,lb,x)
     (ub .- lb) .* x .+ lb
 end
 
+function transform_inf(t , p , f)
+  v(t) = t ./ (1 .- t.^2)
+  if t isa Number
+      j = ForwardDiff.derivative(v, t)
+  else
+      j = det(ForwardDiff.jacobian(x ->v(x), t))
+  end
+  f(v(t) , p)*(j)
+end
+
+function transform_semiinf(t , p , f , lb)
+  v(t) = lb .+ (t ./ (1 .- t))
+  if t isa Number
+      j = ForwardDiff.derivative(v, t)
+  else
+      j = det(ForwardDiff.jacobian(x ->v(x), t))
+  end
+  f(v(t) , p)*(j)
+end
+
+function transformation_if_inf(prob)
+    if (!(prob.lb isa Number) && all(prob.lb .== -Inf) && all(prob.ub .== Inf)) || (prob.lb == -Inf && prob.ub == Inf)
```

Comment:
I wonder if this could be selectively applied to the directions which are Inf.

Reply:
Oh, does the transformation hold for that case?

Reply:
You can cache the indexes to transform and then if transform, then X otherwise identity.